test-case/check-pause-suspend-resume-release.sh

```diff
@@ -107,29 +99,36 @@ esac
 
 [[ -z $file_name ]] && file_name=$dummy_file
 
-
 logger_disabled || func_lib_start_log_collect
 
-setup_kernel_check_point
+func_pipeline_export "$tplg" "type:$test_mode & ${OPT_VAL['p']}"
+
+for idx in $(seq 0 $(expr $PIPELINE_COUNT - 1))
+do
+    channel=$(func_pipeline_parse_value "$idx" channel)
+    rate=$(func_pipeline_parse_value "$idx" rate)
+    fmt=$(func_pipeline_parse_value "$idx" fmt)
+    dev=$(func_pipeline_parse_value "$idx" dev)
+    snd=$(func_pipeline_parse_value "$idx" snd)
 
-dlogi "Entering audio stream expect script with: $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q"
-dlogi "Will enter suspend-resume cycle during paused period of audio stream process"
+    dlogi "Entering audio stream expect script with: $cmd -D $dev -r $rate -c $channel -f $fmt -vv -i $file_name -q"
+    dlogi "Will enter suspend-resume cycle during paused period of audio stream process"
 
-rm -rf /tmp/sof-test.lock
+    rm -rf /tmp/sof-test.lock
 
-# expect is tcl language script
-#   catch: Evaluate script and trap exceptional returns
-#   after ms: Ms must be an integer giving a time in milliseconds.
-#       The command sleeps for ms milliseconds and then returns.
-expect <<AUDIO
-spawn $cmd -D $pcm -r $rate -c $channel -f $fmt -vv -i $dummy_file -q
+    # expect is tcl language script
+    #   catch: Evaluate script and trap exceptional returns
+    #   after ms: Ms must be an integer giving a time in milliseconds.
+    #       The command sleeps for ms milliseconds and then returns.
```

Comment:
If you change the indentation then take that opportunity to use more shell functions, see why in #740 . One of the function should be something like `test_single_pcm()`